ReSharper.FSharp/src/FSharp.Psi.Services/src/Generate/GenerateProvider.fs

```diff
@@ -199,18 +199,17 @@ type FSharpOverridingMembersBuilder() =
         let indentSize = typeDecl.GetIndentSize()
         let desiredIndent = typeDecl.Indent + indentSize
 
-        addNodesBefore typeRepr.FirstChild [
-            NewLine(typeRepr.GetLineEnding())
-            Whitespace(typeRepr.Indent)
-        ] |> ignore
+        let origTypeReprIndent = typeRepr.Indent
+        addNodeBefore typeRepr (NewLine(typeRepr.GetLineEnding()))
+        addNodeBefore typeRepr.FirstChild (Whitespace(origTypeReprIndent))
```

Comment:
This still (likely) adds a child to a wrong place. Please check the tree structure with a parser test.

Why likely? There's some logic in R# that tries to move whitespace nodes (e.g. `Whitespace`, `NewLine`) to parent nodes when they are at a node boundary (i.e. a node shouldn't start/end with a whitespace). So there's a chance that this logic fixes the tree structure for you, but the modification here itself looks wrong.

Reply:
The dump looks correct, all seems good now!